app/public/lobby/lobbyfuncs.js

```diff
@@ -4,62 +4,93 @@ const refreshBtn = document.getElementById("refresh");
 
 const secretBtn = document.getElementsByClassName("header-btn")[0]; // for testing
 
-// will get this working with server and actual requirements later, just playing around for now
-let gameHostName = 1;
-let gameList = [];
+
+const username = Math.floor(Math.random()*10000);
+
+// move this stuff to backend later
+let lobbyId = 1;
+let joinedLobbyId = 0;
 
 // leaving table as a parameter for now
+// values: [lobbyId, hostname, playerNum, timeCreated]
 const addRowToTopOfTable = (table, values) => {
 	let gameRow = document.createElement("tr");
-	gameRow.id = values[0]; // just use host name for now
-	for (let i=0; i<values.length; i++) {
+	gameRow.id = values[0]; // id = lobbyId for now
+	for (let i=1; i<values.length; i++) {
 		let cell = document.createElement("td");
 		cell.innerText = values[i];
 		gameRow.append(cell);
 	}
 	gameRow.addEventListener("click", () => {
+		joinedLobbyId = values[0];
 		post(
 			"/lobby/join",
-			{"lobbyId" : values[0], "username" : username}
+			{"lobbyId" : joinedLobbyId, "username" : username}
 		).then((resp) => {
 			if (resp.status !== 200) {
 				// TO DO: Handle errors in joining lobby when lobby system is done
 				console.log("ERROR - Something went wrong");
 			}
-			// add row to TOP of lobby list
 		}).catch((e) => {
 			console.log(e);
 		});
+		
 	});
 	lobbyTbl.prepend(gameRow);
 }
 
 createBtn.addEventListener("click", () => {
-	/*
-	post(
-		"/lobby/make",
-		{}
+	if (joinedLobbyId === 0) {
+		getValidId();
+		post(
+			"/lobby/make",
+			{"lobbyId" : lobbyId, "username" : username}
+		).then((resp) => {
+			if (resp.status !== 200) {
+				// TO DO: Handle errors in joining lobby when lobby system is done
+				console.log("ERROR - Something went wrong");
+			}
+			else {
+				joinedLobbyId = lobbyId;
+				lobbyId++;
+			}
+			// table should update shortly
+		}).catch((e) => {
+			console.log(e);
+		});
+	}
+	else {
+		console.log("stop spamming!");
+		// to avoid weird effects when one client makes a bunch of lobbies
+		// temporary
+	}
+	
+});
+
+// lobby refreshes automatically, so this is only needed if a lobby is removed
+// any other buttons we should have?
+refreshBtn.addEventListener("click", () => {
+	fetch(
+		"/lobby/list"
 	).then((resp) => {
 		if (resp.status !== 200) {
 			// TO DO: Handle errors in joining lobby when lobby system is done
 			console.log("ERROR - Something went wrong");
 		}
-		// add row to TOP of lobby list
+		// update table according to list
+		resp.json().then((lobbyList) => {
+			console.log(lobbyList);
+			refreshLobbies(lobbyList);
+		});
```

Comment:
Long said every promise needs a catch ;)
Seems a little tedious but it can't hurt